.github/workflows/security.yaml

```diff
@@ -32,7 +35,60 @@ jobs:
           api-key: ${{ secrets.FOSSA_API_KEY }}
 
       - name: Run FOSSA Test
-        uses: fossas/fossa-action@main
-        with:
-          api-key: ${{ secrets.FOSSA_API_KEY }}
-          run-tests: true
+        id: fossa_test
+        env:
+          FOSSA_API_KEY: ${{ secrets.FOSSA_API_KEY }}
+        run: |
+          MAIN_BRANCH="refs/heads/main"
+          CURRENT_BRANCH="${{ github.ref }}"
+          DIFF_OPTION=""
+      
+          if [ "$CURRENT_BRANCH" != "$MAIN_BRANCH" ]; then
+            git fetch origin main
+            MAIN_SHA=$(git rev-parse origin/main)
+            DIFF_OPTION="--diff $MAIN_SHA"
+          fi
+      
+          fossa test --format json $DIFF_OPTION | tee fossa-test.json 1>&2
```

Comment:
in case you want to do any JSON parsing or formatting, I recommend [`jq`](https://github.com/bobbyiliev/introduction-to-bash-scripting/blob/main/ebook/en/content/018-working-with-json-in-bash-using-jq.md) (I may be telling something you already know, just sharing the love)

Reply:
oh lol you literally use `jq` a few lines down